githubpoxy.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"errors"
+	"io"
+	"time"
+
+	"github.com/go-rod/rod"
+)
+
+type GitHubProvider struct {
+}
+
+func (ghp GitHubProvider) Login(login, password string) error {
+	const loginURL = "https://github.com/login"
+
+	// Open page
+	browser := rod.New()
+	err := browser.Connect()
+
+	if err != nil {
+		return errors.New("Unable to connect to broswer")
+	}
+
+	defer browser.MustClose()
+	browser.ControlURL(loginURL)
+	err = browser.Connect()
+	if err != nil {
+		return errors.New("Unable to open GitHub login URL")
+	}
+	page := browser.MustPage(loginURL)
+
+	// Enter credentials
+	loginElement, err := page.Element("#login_field")
+
+	if err != nil {
+		return errors.New("Unable to find login text field")
+	}
+
+	passwordElement, err := page.Element("#password")
+
+	if err != nil {
+		return errors.New("Unable to find password text field")
+	}
+
+	loginElement.MustClick().MustInput(login)
+	passwordElement.MustClick().MustInput(password)

```

Comment:
Try to avoid `panic`-ing in library functions. There's a `rod.Try` function for quick prototyping but ideally we'd want conventional `if err != nil` style checking in this function.